fulltext/backends/__json.py

```diff
@@ -0,0 +1,32 @@
+import json
+
+from six import StringIO
+from six import string_types
+from six import integer_types
+
+
+def _to_text(text, obj):
+    if isinstance(obj, dict):
+        for key in sorted(obj.keys()):
+            _to_text(text, key)
+            _to_text(text, obj[key])
+
+    elif isinstance(obj, list):
+        for item in obj:
+            _to_text(text, item)
+
+    elif isinstance(obj, string_types):
+        text.write(obj)
+        text.write(u' ')
+
+    elif isinstance(obj, integer_types):
+        text.write(str(obj))
+        text.write(u' ')
+
+
+def _get_file(f, **kwargs):
+    text, obj = StringIO(), json.loads(f.read().decode('utf8'))
```

Comment:
I find this a bit hard to read (mainly because json.loads has many parentheses)

Reply:
Not sure how JSON handles unicode or whether there's a problem here. Just thinking we'll need unit tests for non-ASCII text in which case we may get a failure here.

Reply:
This is a concern, the problem is that the file was written using some encoding. Without knowing the encoding, I can't know what codec to use. I guess at `utf8` as it is very common. However, in the text backend, I use the trick of reading the encoding from `sys.getfilesystemencoding()`, and also accepting an encoding parameter from the user (via kwargs). I should probably do the same wherever I need a codec.

Reply:
https://github.com/btimby/fulltext/blob/master/fulltext/backends/__text.py#L23